Blocks/src/Controller/Admin/BlocksController.php

```diff
@@ -121,23 +121,25 @@ public function index() {
  * @return void
  * @access public
  */
-	public function admin_add() {
+	public function add() {
 		$this->set('title_for_layout', __d('croogo', 'Add Block'));
 
+		$block = $this->Blocks->newEntity();
 		if (!empty($this->request->data)) {
-			$this->Block->create();
-			$this->request->data['Block']['visibility_roles'] = $this->Block->encodeData($this->request->data['Role']['Role']);
-			$this->request->data['Block']['visibility_paths'] = $this->Block->encodeData(explode("\n", $this->request->data['Block']['visibility_paths']));
-			if ($this->Block->saveAll($this->request->data)) {
-				$this->Session->setFlash(__d('croogo', 'The Block has been saved'), 'flash', array('class' => 'success'));
-				$this->Croogo->redirect(array('action' => 'edit', $this->Block->id));
+			$this->request->data['visibility_roles'] = $this->Blocks->encodeData($this->request->data['Role']['Role']);
+			$this->request->data['visibility_paths'] = $this->Blocks->encodeData(explode("\n", $this->request->data['Block']['visibility_paths']));
```

Comment:
We've made a special database type that should be able to handle the encoding and decoding of data, its located in https://github.com/croogo/croogo/blob/3.0/Core/src/Database/Type/ParamsType.php